review-skills crashes with raw fjall lock error when server is running

## Summary

\`aletheia review-skills --nous-id chiron --action list\` crashes with:
\`\`\`
Error: failed to open knowledge store
Caused by: FjallError: Locked
\`\`\`

Other \`memory\` subcommands (\`sample\`, \`consolidate\`, \`dedup\`, \`patterns\`) show a friendly message telling the user to stop the server. \`memory check\` transparently routes through the API.

\`review-skills\` should either route through the API like \`memory check\`, or show the same friendly lock message.

## Environment

- aletheia 0.17.0, fjall session store

--- a/crates/aletheia/src/commands/agent_io.rs
+++ b/crates/aletheia/src/commands/agent_io.rs
@@ -80,6 +80,9 @@ pub(crate) struct ExportSkillsArgs {
     /// Filter by domain tags (comma-separated)
     #[arg(short, long)]
     pub domain: Option<String>,
+    /// Server URL for lock detection
+    #[arg(long, default_value = "http://127.0.0.1:18789")]
+    pub url: String,
 }
 
 #[derive(Debug, Clone, Args)]
@@ -513,10 +516,11 @@ pub(crate) fn seed_skills(args: &SeedSkillsArgs) -> Result<()> {
 ///
 /// Reads skill facts from an in-process `KnowledgeStore`, converts them to
 /// `SkillContent`, and writes `.claude/skills/<slug>/SKILL.md` files.
-pub(crate) fn export_skills(
+pub(crate) async fn export_skills(
     instance_root: Option<&PathBuf>,
     args: &ExportSkillsArgs,
 ) -> Result<()> {
+    guard_knowledge_lock(&args.url).await?;
     #[cfg(feature = "recall")]
     {
         use mneme::knowledge_store::KnowledgeStore;
@@ -781,7 +785,7 @@ fn generate_simple_embedding(text: &str) -> Vec<f32> {
 ///
 /// Returns an error with a helpful message if the server is reachable,
 /// preventing a confusing `FjallError::Locked` crash.
-async fn guard_knowledge_lock(url: &str) -> Result<()> {
+pub(crate) async fn guard_knowledge_lock(url: &str) -> Result<()> {
     let endpoint = format!("{url}/api/health");
     if let Ok(resp) = reqwest::get(&endpoint).await
         && (resp.status().is_success() || resp.status().as_u16() == 503)

--- a/crates/aletheia/src/commands/mod.rs
+++ b/crates/aletheia/src/commands/mod.rs
@@ -98,7 +98,7 @@ pub(crate) async fn dispatch(cmd: Command, instance_root: Option<&PathBuf>) -> R
         Command::Import(a) => agent_io::import_agent(instance_root, &a).map_err(Into::into),
         Command::SeedSkills(a) => agent_io::seed_skills(&a).map_err(Into::into),
         Command::ExportSkills(a) => {
-            agent_io::export_skills(instance_root, &a).map_err(Into::into)
+            agent_io::export_skills(instance_root, &a).await.map_err(Into::into)
         }
         Command::ReviewSkills(a) => {
             agent_io::review_skills(instance_root, &a).await.map_err(Into::into)
@@ -113,7 +113,7 @@ pub(crate) async fn dispatch(cmd: Command, instance_root: Option<&PathBuf>) -> R
         Command::CheckConfig => check_config::run(instance_root).map_err(Into::into),
         Command::Config { action } => config::run(&action, instance_root).map_err(Into::into),
         Command::AddNous(a) => add_nous::run(instance_root, &a).await.map_err(Into::into),
-        Command::Repl(a) => repl::run(instance_root, &a).map_err(Into::into),
+        Command::Repl(a) => repl::run(instance_root, &a).await.map_err(Into::into),
         // NOTE: Serve is intercepted in main() before dispatch is called.
         // This arm exists only for match exhaustiveness.
         #[expect(

--- a/crates/aletheia/src/commands/repl.rs
+++ b/crates/aletheia/src/commands/repl.rs
@@ -12,7 +12,9 @@ use crate::error::Result;
 
 #[derive(Debug, Clone, Args)]
 pub(crate) struct ReplArgs {
-    // instance_root is inherited from the top-level -r flag; nothing extra needed.
+    /// Server URL for lock detection
+    #[arg(long, default_value = "http://127.0.0.1:18789")]
+    pub url: String,
 }
 
 #[cfg(feature = "recall")]
@@ -32,10 +34,12 @@ Datalog examples:
 
 /// Run the interactive Datalog REPL.
 ///
-/// WHY: Synchronous stdin loop — async is unnecessary here and would add
-/// complexity. The knowledge store holds an exclusive fjall lock, so the
-/// server must be stopped before starting the REPL.
-pub(crate) fn run(instance_root: Option<&PathBuf>, _args: &ReplArgs) -> Result<()> {
+/// WHY: the guard check is async (HTTP health probe), but the REPL loop itself
+/// is synchronous stdin I/O. We check for a running server first, then drop
+/// into the blocking loop.
+pub(crate) async fn run(instance_root: Option<&PathBuf>, args: &ReplArgs) -> Result<()> {
+    super::agent_io::guard_knowledge_lock(&args.url).await?;
+
     #[cfg(feature = "recall")]
     {
         run_repl(instance_root)

